pkgs/development/web/rover/default.nix

```diff
@@ -0,0 +1,62 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, perl
+, openssl
+, pkg-config
+, darwin
+, stdenv
+}:
+
+let
+  # The build fetches this dynamically from https://graphql.api.apollographql.com/api/schema
+  apolloSchema = builtins.readFile ./schema.graphql;
+  # The build fetches this from the headers of https://graphql.api.apollographql.com/api/schema
+  etagId = builtins.readFile ./etag.id;
+in
+rustPlatform.buildRustPackage rec {
+  pname = "rover";
+  version = "0.1.10";
+
+  src = fetchFromGitHub {
+    owner = "apollographql";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0vmm35b9d7xw160pn797z78vj8jaav0qb6dw7n4ld8ssj076dsrh";
+  };
+
+  nativeBuildInputs = [ perl pkg-config ];
+  buildInputs = [ openssl ] ++ lib.optional stdenv.isDarwin darwin.apple_sdk.frameworks.Security;
+
+  # Put schema in place that would otherwise be dynamically fetched during the build
+  preBuildPhase = [
+    ''
+      mkdir -p crates/rover-client/.schema
+      cat $apolloSchemaPath > crates/rover-client/.schema/schema.graphql
+      echo ${etagId} > crates/rover-client/.schema/etag.id
+    ''
+  ];
+  preBuildPhases = [ "preBuildPhase" ];
```

Comment:
```suggestion
  preBuild = ''
    mkdir -p crates/rover-client/.schema
    cp $apolloSchemaPath crates/rover-client/.schema/schema.graphql
    echo ${etagId} > crates/rover-client/.schema/etag.id
  ''
```

Reply:
:+1: thanks, changed in 41801e346d0175372c906f75c782eb20e706199f